node/src/neighborhood/mod.rs

```diff
@@ -118,29 +118,30 @@ impl Handler<BindMessage> for Neighborhood {
     fn handle(&mut self, msg: BindMessage, ctx: &mut Self::Context) -> Self::Result {
         ctx.set_mailbox_capacity(NODE_MAILBOX_CAPACITY);
         self.hopper_opt = Some(msg.peer_actors.hopper.from_hopper_client);
-        self.hopper_no_lookup_opt = Some(msg.peer_actors.hopper.from_hopper_client_no_lookup);
-        self.connected_signal_opt = Some(msg.peer_actors.accountant.start);
-        self.gossip_acceptor_info_opt = Some (Either::Right (GossipAcceptorSubs {
+        self.hopper_no_lookup_sub_opt = Some(msg.peer_actors.hopper.from_hopper_client_no_lookup);
+        self.connected_signal_sub_opt = Some(msg.peer_actors.accountant.start);
+        self.gossip_acceptor_info_opt = Some(Either::Right(GossipAcceptorSubs {
             connection_progress_sub: msg.peer_actors.neighborhood.connection_progress_sub,
             stream_shutdown_sub: msg.peer_actors.dispatcher.stream_shutdown_sub,
         }));
         self.gossip_producer_opt = Some(Box::new(GossipProducerReal::new()));
-        self.node_to_ui_recipient_opt = Some(msg.peer_actors.ui_gateway.node_to_ui_message_sub);
+        self.node_to_ui_sub_opt = Some(msg.peer_actors.ui_gateway.node_to_ui_message_sub);
     }
 }
 
 impl Handler<PoolBindMessage> for Neighborhood {
     type Result = ();
 
     fn handle(&mut self, msg: PoolBindMessage, _: &mut Self::Context) -> Self::Result {
-        let gossip_acceptor_subs = self.gossip_acceptor_info_opt
+        let gossip_acceptor_subs = self
+            .gossip_acceptor_info_opt
             .take()
             .expect("Neighborhood never got its BindMessage")
-            .right_or_else (|_| panic! ("Neighborhood got multiple PoolBindMessages"));
-        self.gossip_acceptor_info_opt = Some(Either::Left (Box::new(GossipAcceptorReal::new(
+            .right_or_else(|_| panic!("Neighborhood got multiple PoolBindMessages"));
```

Comment:
I remember that you once exclaimed victoriously that the need for this complexity with different stages in the long initialization of the `gossip_acceptor` was gone. 
I was probably mistaken then but yes I'm surprised to see it still so bright here in this place.   

Speaking of the Either referred about both on the line above and below.

Reply:
I think that was a change that Utkarsh made. It's either in `master` or will be there soon; I'll eventually have to merge it in.